### ISSUE TITLE
Programmatically creating facelets with @View doesn't work

### ISSUE THREAD

**Issue by simasch** [melloware/quarkus-faces#444]
I tried to add a programmatic view with the `@View` annotation, but when I access http://localhost:8080/facelet.xhml it returns 404

```
@View("/facelet.xhtml")
@ApplicationScoped
public class FaceletView extends Facelet {

    @Override
    public void apply(FacesContext facesContext, UIComponent parent) {
        if (!facesContext.getAttributes().containsKey(IS_BUILDING_INITIAL_STATE)) {
            return;
        }

        var components = new ComponentBuilder(facesContext);
        var rootChildren = parent.getChildren();

        var doctype = new UIOutput();
        doctype.setValue("<!DOCTYPE html>");
        rootChildren.add(doctype);

        var htmlTag = new UIOutput();
        htmlTag.setValue("<html xmlns=\"http://www.w3.org/1999/xhtml\">");
        rootChildren.add(htmlTag);

        HtmlBody body = components.create(HtmlBody.COMPONENT_TYPE);
        rootChildren.add(body);

        HtmlForm form = components.create(HtmlForm.COMPONENT_TYPE);
        form.setId("form");
        body.getChildren().add(form);

        HtmlOutputText message = components.create(HtmlOutputText.COMPONENT_TYPE);
        message.setId("message");

        HtmlCommandButton actionButton = components.create(HtmlCommandButton.COMPONENT_TYPE);
        actionButton.setId("button");
        actionButton.addActionListener(
                e -> message.setValue("Hello, World! Welcome to Faces 4.0 on Jakarta EE 10"));
        actionButton.setValue("Greet");

        form.getChildren().add(actionButton);

        parent.getChildren().add(message);

        htmlTag = new UIOutput();
        htmlTag.setValue("</html>");
        rootChildren.add(htmlTag);
    }

    private static class ComponentBuilder {
        FacesContext facesContext;

        ComponentBuilder(FacesContext facesContext) {
            this.facesContext = facesContext;
        }

        @SuppressWarnings("unchecked")
        <T> T create(String componentType) {
            return (T) facesContext.getApplication().createComponent(facesContext, componentType, null);
        }
    }
}

```

**Comment by melloware** [comment]
I wonder if nobody has tried this yet and the Quarkus MyFaces extension is not scanning for `@View`.  I will investigate.

**Comment by melloware** [comment]
Standalone Jetty runnner of the example above just so i could make sure it works outside of Quarkus.
[quarkus-444.zip](https://github.com/melloware/quarkus-faces/files/15475931/quarkus-444.zip)

`mvn clean jetty:run` and navigate to http://localhost:8080/facelet.xhtml

**Comment by melloware** [comment]
I opened a MyFaces Quarkus Extension ticket: https://issues.apache.org/jira/browse/MYFACES-4668

**Comment by melloware** [comment]
Also opened this: https://issues.apache.org/jira/browse/MYFACES-4669

**Comment by melloware** [comment]
I have a fix for Quarkus but waiting to hear back on the escaping issue with the UIOutput

**Comment by melloware** [comment]
ok for MyFaces I had to update your example to be this using `DocType` and removing the `<html>` tag.

```java
package org.primefaces.test;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.faces.annotation.View;
import jakarta.faces.application.StateManager;
import jakarta.faces.component.UIComponent;
import jakarta.faces.component.html.HtmlBody;
import jakarta.faces.component.html.HtmlCommandButton;
import jakarta.faces.component.html.HtmlDoctype;
import jakarta.faces.component.html.HtmlForm;
import jakarta.faces.component.html.HtmlOutputText;
import jakarta.faces.context.FacesContext;
import jakarta.faces.view.facelets.Facelet;

@View("/facelet.xhtml")
@ApplicationScoped
public class FaceletView extends Facelet {

    @Override
    public void apply(FacesContext facesContext, UIComponent parent) {
        if (!facesContext.getAttributes().containsKey(StateManager.IS_BUILDING_INITIAL_STATE)) {
            return;
        }

        var components = new ComponentBuilder(facesContext);
        var rootChildren = parent.getChildren();

        var htmlDoctype = new HtmlDoctype();
        htmlDoctype.setRootElement("html");
        rootChildren.add(htmlDoctype);

        HtmlBody body = components.create(HtmlBody.COMPONENT_TYPE);
        rootChildren.add(body);

        HtmlForm form = components.create(HtmlForm.COMPONENT_TYPE);
        form.setId("form");
        body.getChildren().add(form);

        HtmlOutputText message = components.create(HtmlOutputText.COMPONENT_TYPE);
        message.setId("message");

        HtmlCommandButton actionButton = components.create(HtmlCommandButton.COMPONENT_TYPE);
        actionButton.setId("button");
        actionButton.addActionListener(
                e -> message.setValue("Hello, World! Welcome to Faces 4.0 on Jakarta EE 10"));
        actionButton.setValue("Greet");

        form.getChildren().add(actionButton);

        body.getChildren().add(message);
    }

    private static class ComponentBuilder {
        FacesContext facesContext;

        ComponentBuilder(FacesContext facesContext) {
            this.facesContext = facesContext;
        }

        @SuppressWarnings("unchecked")
        <T> T create(String componentType) {
            try {
                return (T) facesContext.getApplication().createComponent(componentType);
            } catch (ClassCastException e) {
                throw new IllegalArgumentException("Component type " + componentType + " is not valid.", e);
            }
        }
    }
}
```

**Comment by melloware** [comment]
I also asked on Mojarra which is the correct behavior https://github.com/eclipse-ee4j/mojarra/issues/5452

**Comment by melloware** [comment]
@simasch i added this fix in Quarkus PrimeFaces 3.14.6.1 so we don't have to wait for a MyFaces release: https://github.com/quarkiverse/quarkus-primefaces/blob/804d53a3bd56d88c6077775f36db145d22377ba9/quarkus-primefaces/deployment/src/main/java/io/quarkiverse/primefaces/deployment/PrimefacesProcessor.java#L177-L197

**Comment by simasch** [comment]
Thank you @melloware
